### PR TITLE
Added: UIElementButton

### PR DESCRIPTION
**Pull request by KonWol**
### Type of change

* [ ] Bug fix
* [x] New feature
* [ ] Breaking change
* [ ] Maintenance (eg., documentation, code cleanup)

## General information

|Developed and tested on:                                         ||
|--------------------------------|--------------------------------|
|Web browser incl. exact version | Chromium: 74.0.3729.131 (Official Build)|
|Operating system                |Arch Linux (64-Bit)|

* [ ] Web browser/platform specific
  If yes: ...

## What does the changes do?

Implemented a `UIElementButton(className, caption, actionFunction)` where `actionFunction` is a callback function (can be anonymous) that is executed.

## (optional) What is the benefit?

...

## Final checks

* [x] Commit messages are explanatory  
  (for examples see [commits](https://github.com/TheFragebogen/TheFragebogen/commits/master))
* [x] Code follows [CODING.md](CODING.md)
* [x] (optional) executed `grunt format; grunt lint; grunt test`

## License
By submitting a pull request, I understand that the code is provided under [MIT License](LICENSE.md).

### CODE REVIEW

**Comment by KonWol** [comment]
I thought about implementing a counter for statistics on how often the button got pressed. But this seems not to be possible with UIElements. Am I right with this?

**Comment by dennisguse** [comment]
Data export is only implemented for Screens and QuestionnaireItems. It is implemented in ScreenController.requestDataArray() and ScreenUIElements.getDataCSV()

File: src/uielement_button.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+*/
+function UIElementButton(className, caption, actionFunction) {
+    UIElement.call(this);
+
+    this.className = className;
+    this.caption = caption;
+    this.actionFunction = actionFunction;
+
+    this.node = null;
+    this.button = null;
```
Comment (dennisguse) [inline src/uielement_button.js:20]: button is never used outside of createUI() and thus not required to be an attribute.

File: src/uielement_button.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+The provided callback function gets called if a onclick event occurs on this button.
+
+@class UIElementButton
+@augments UIElement
+
+@param {string} [className] CSS class
+@param {string} caption Caption of the button
+@param {method} actionFunction Callback function for onclick event
+*/
+function UIElementButton(className, caption, actionFunction) {
```
Comment (dennisguse) [inline src/uielement_button.js]: @actionFunction: you are right - it is a callback (as stated in line 3).
Better name would be actionCallback.

**Review by dennisguse — CHANGES_REQUESTED** [review]
Minor things.

File: src/uielement_button.js (hunk 1 above)
Comment (KonWol) [reply src/uielement_button.js:20]: `setEnabled()` uses the button. Whether it is necessary to implement `setEnabled()`, I don't know.

File: src/uielement_button.js (hunk 2 above)
Comment (KonWol) [reply src/uielement_button.js]: Fixed.

**Comment by dennisguse** [comment]
You are right - I didnt see the reference.
Set enabled so far used this.node instead of disabling all 
HTMLComponents individually.
Then you can avoid this.button.

About setEnabled(): It is used by ScreenUIElementsSequential.
It uses setEnabled() to activate UIElements one after the other 
(actually build for QuestionnaireItems).
However, it a UIElement would need to inherit from UIElementInteractive, 
so that the event callbacks are there.
Otherwise ScreenUIElementsSequential won't react to any change.
In any case, I made the mistake to recommend inheriting from UIElement.
Could you adjust this?

I will create a demo for ScreenUIElementsSequential and also *Auto next 
week.
I just realized, it was never implemented.

**Comment by KonWol** [comment]
I would suggest inheriting the button from QuestionnaireItem. This would make the following things easier:
- implement that a button might have to be `required` to press
-- `actionCallback` can decide if the requirements are met by returning `false` or `true`
- provide a metric of how often the button was pressed

What du you think about that?

**Comment by dennisguse** [comment]
I know what you mean. However, inheriting from QuestionnaireItem also adds creation of a question and that is something not necessary. What could be done is moving the required code to UiElementInteractive. 

Providing a metric is the same thing. We could enable data export via an interface (ie function) that is used if present.

However, I would postpone both issues.
There is an additional issue here: we for the first time connect two different elements and also these cannot communicate their state. For replay, it would actually be be better to add this functionality to the QuestionnaireItemMediaVideo (or a child). Then all UI belongs to the same managing class.
Just overwrite createUI and add the necessary UI components there. However, calling the super method is only available after upgrade to ES6.

Long story short: lets leave it as it is.
And improve the features if required.
You can also add an issue, so we want forget about it.
